lens/lily/impl.go

```diff
@@ -113,23 +163,44 @@ func (m *LilyNodeAPI) LilyWatch(_ context.Context, cfg *LilyWatchConfig) (*sched
 		JobName: cfg.Name,
 	}
 
-	// create a database connection for this watch, ensure its pingable, and run migrations if needed/configured to.
-	strg, err := m.StorageCatalog.Connect(ctx, cfg.Storage, md)
-	if err != nil {
-		return nil, err
+	wapi := &watcherAPIWrapper{
+		Events:         m.Events,
+		ChainModuleAPI: m.ChainModuleAPI,
 	}
 
-	taskAPI, err := datasource.NewDataSource(m)
-	if err != nil {
-		return nil, err
-	}
+	var idx indexer.Indexer
+	if cfg.Queue == "" {
```

Comment:
IMO, this condition is very mysterious at first glance and isn't clear why this should trigger the storage setup vs not. If possible, I would make it more clear through better variable/function names or comments.

Something like `cfg.IsNotifier() bool` or `configureForNotifier(*LilyWatchConfig) bool`? Or whatever functional behavior this is actually checking for. (ex: IsNotifier describes behavior vs IsIntegrated which describes the domain object. For the casual reader, the behavioral name is more valuable of the two.)

Reply:
Or BETTER, make this a concern inside of `NewWatcher`. I don't see any reason for setup to live in this abstraction when so many of these are args to NewWatcher anyway. WDYT?

Reply:
Yeah great point, I wasn't sure how to wire this all up. 

How would you feel about new methods that are specific to the distributed indexer? Something like `WatchNotify`, `WalkNotify`, `FillNotify`.

Reply:
This seems like a good idea. 👍